sim/src/main.rs

```diff
@@ -38,17 +34,25 @@ fn main() {
     // Create simulator object.
     let mut simulator: Simulator = Simulator::new();
 
-    let tgen = simulator.add_node(TrafficGenerator::new(1, 0));
-    let node1 = simulator.add_node(Link::new(2, 1, plugin_str, 1));
-    let node2 = simulator.add_node(Link::new(2, 1, plugin_str, 2));
-    let node3 = simulator.add_node(Link::new(2, 1, plugin_str, 3));
-    let node4 = simulator.add_node(Link::new(2, 1, plugin_str, 4));
-
-    let _edge5 = simulator.add_one_direction_edge(Channel::new(1, 5), tgen, node1);
-    let _edge6 = simulator.add_edge(Channel::new(2, 6), node1, node2);
-    let _edge7 = simulator.add_edge(Channel::new(2, 7), node1, node3);
-    // one way rpc sink
-    let _edge8 = simulator.add_one_direction_edge(Channel::new(1, 8), node1, node4);
+    // node arguments go:  id, capacity, egress_rate, generation_rate, plugin
+    simulator.add_node("traffic generator", 10, 1, 1, plugin_str);
+    simulator.add_node("node 1", 10, 1, 0, plugin_str);
+    simulator.add_node("node 2", 10, 1, 0, plugin_str);
+    simulator.add_node("node 3", 10, 1, 0, plugin_str);
+    simulator.add_node("node 4", 10, 1, 0, plugin_str);
+
+    // edge arguments go:  delay, endpoint1, endpoint2, unidirectional
+    simulator.add_edge(
+        1,
+        "tgen->node1",
```

Comment:
Is that the edge name? I think that should be automatically inferred, or made optional. Seems unnecessary to include it in the base api